src/meta/src/controller/catalog/drop_op.rs

```diff
@@ -15,6 +15,76 @@
 use super::*;
 
 impl CatalogController {
+    // drop table associated source is a special case of drop relation, which just remove the source object and associated state table, keeping the streaming job and fragments.
+    pub async fn drop_table_associated_source(
+        &self,
+        txn: &DatabaseTransaction,
+        release_ctx: &ReleaseContext,
```

Comment:
Using `ReleaseContext` as the parameter here is very confusing for me. The target table will not be removed, but updated instead. Can we use three distinct parameters here?

Reply:
The table_id here is the corresponding state table's id, it is to be removed. I think that's the right semantic?

Reply:
Yes, the `removed_state_table_ids` is the right semantic, but `removed_streaming_job_ids` is not.